### DISCUSSION TITLE
.fill modifier of x-model directive is not working properly when using on HTML select element

### DISCUSSION THREAD

**Discussion by tjodalv** [alpinejs/alpine#3554] [7. Bugs (archived)] [answered]
Hi

I would like to report a bug on latest version of Alpine, currently it is `3.12.0`. I am using x-model directive on `<select>` element and would like to fill binded property with the value of currently selected option, if binded property is null or empty string. For that purpose there is `.fill` modifier but that modifier doesn't work correctly with `<select>` element. Consider this example:

```html
<select x-data="{ selected: '' }" x-model.fill="selected">
    <option value="1" selected>One</option>
    <option value="2">Two</option>
    <option value="3">Three</option>
</select>
```

In this example, binded property named `selected` will not be populated with the selected value. If I change the value by selecting other option then the change is reflected, but not on page load. So, instead of the `.fill` modifier, for `<select>` elements, I have to do this:

```html
<select x-data="{ selected: '' }" x-model="selected" x-init="selected=$el.value">
    <option value="1" selected>One</option>
    <option value="2">Two</option>
    <option value="3">Three</option>
</select>
```

I have tried to use x-model.fill modifier on text element and there it works as expected, but it is not working for the `<select>` elements. Didn't try if there is the same problem for other non-text input types such as checkbox or radio buttons.

**Comment by SimoTod** [comment, marked as answer]
https://github.com/alpinejs/alpine/pull/3495
